assert/assertions.go

```diff
@@ -357,9 +357,11 @@ func formatUnequalValues(expected, actual interface{}) (e string, a string) {
 		return fmt.Sprintf("%T(%#v)", expected, expected),
 			fmt.Sprintf("%T(%#v)", actual, actual)
 	}
-
-	return fmt.Sprintf("%#v", expected),
-		fmt.Sprintf("%#v", actual)
+	switch expected.(type) {
```

Comment:
Any reason for preferring a `switch` over a plain `if`? Would it make sense to have it as an:
```
if expected.(type) == time.Duration {
    return ...
}
return ...
```
and if we need to add more cases in the future, we can convert to a `switch`?

Reply:
type check needs switch